pkg/softtainter/softtainter.go

```diff
@@ -0,0 +1,532 @@
+package softtainter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	promapi "github.com/prometheus/client_golang/api"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/pkg/util/taints"
+
+	desv1 "github.com/openshift/cluster-kube-descheduler-operator/pkg/apis/descheduler/v1"
+	"github.com/openshift/cluster-kube-descheduler-operator/pkg/operator/operatorclient"
+	"sigs.k8s.io/descheduler/pkg/api"
+	"sigs.k8s.io/descheduler/pkg/descheduler"
+	desclient "sigs.k8s.io/descheduler/pkg/descheduler/client"
+	nodeutil "sigs.k8s.io/descheduler/pkg/descheduler/node"
+	"sigs.k8s.io/descheduler/pkg/framework/pluginregistry"
+	"sigs.k8s.io/descheduler/pkg/framework/plugins/nodeutilization"
+	"sigs.k8s.io/descheduler/pkg/framework/plugins/nodeutilization/classifier"
+	"sigs.k8s.io/descheduler/pkg/framework/plugins/nodeutilization/normalizer"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	prometheusAuthTokenSecretKey        = "prometheusAuthToken"
+	OverUtilizedSoftTaintKey            = "nodeutilization.descheduler.kubernetes.io/overutilized"
```

Comment:
nodeutilization.descheduler.kubernetes.io -> nodeutilization.descheduler.sigs.k8s.io

Reply:
Compared to https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions we might use `nodeutilization.descheduler.sigs.kubernetes.io`?

Reply:
Or, given this is a TP/beta feature what about `nodeutilization.descheduler.beta.kubernetes.io`?

Reply:
Also, given this is an openshift specific taint we might go with `nodeutilization.descheduler.beta.openshift.io`?

Reply:
Version dependent taint names will require a leftover cleanup step on upgrades.
If we don't see any good reason to make them version specific, I'd suggest to go with `nodeutilization.descheduler.openshift.io`.

Reply:
I see. Let's go with `nodeutilization.descheduler.openshift.io` then.

Reply:
Done